riscv/processor.cc

```diff
@@ -1153,6 +1521,33 @@ reg_t processor_t::get_csr(int which)
       if (!supports_extension('S'))
         break;
       return state.mideleg;
+    case CSR_HSTATUS: return state.hstatus;
+    case CSR_HEDELEG: return state.hedeleg;
+    case CSR_HIDELEG: return state.hideleg;
+    case CSR_HIE: return state.mie & MIP_HS_MASK;
+    case CSR_HCOUNTEREN: return state.hcounteren;
+    case CSR_HGEIE: return 0;
+    case CSR_HTVAL: return state.htval;
+    case CSR_HIP: return state.mip & MIP_HS_MASK;
+    case CSR_HVIP: return state.mip & MIP_VS_MASK;
+    case CSR_HTINST: return state.htinst;
+    case CSR_HGATP: return state.hgatp;
+    case CSR_HGEIP: return 0;
+    case CSR_VSSTATUS: {
+      reg_t mask = SSTATUS_VS_MASK;
+      mask |= (supports_extension('F') ? SSTATUS_FS : 0);
+      mask |= (supports_extension('V') ? SSTATUS_VS : 0);
+      mask |= (xlen == 64 ? SSTATUS64_SD : SSTATUS32_SD);
+      return state.vsstatus & mask;
+    }
+    case CSR_VSIE: return (state.mie & MIP_VS_MASK) >> 1;
```

Comment:
hideleg, per 5.2.11:
```
When bit 10 of hideleg is zero, vsip.SEIP and vsie.SEIE are read-only zeros. Else, vsip.SEIP
and vsie.SEIE are aliases of hip.VSEIP and hie.VSEIE.
When bit 6 of hideleg is zero, vsip.STIP and vsie.STIE are read-only zeros. Else, vsip.STIP
and vsie.STIE are aliases of hip.VSTIP and hie.VSTIE.
When bit 2 of hideleg is zero, vsip.SSIP and vsie.SSIE are read-only zeros. Else, vsip.SSIP and
vsie.SSIE are aliases of hip.VSSIP and hie.VSSIE.
```

Reply:
Same as above, I should have considered HIDELEG when reading CSR_VSIE and CSR_VSIP.

I will update.